lib/galaxy/webapps/galaxy/api/biotools.py

```diff
@@ -0,0 +1,299 @@
+    :rtype: string
+    """
+    try:
+        size = len(description)
+        if description[size - 1] == '.':
+            return description[0].upper() + description[1:size]
+        else:
+            return description[0].upper() + description[1:size] + '.'
+    except IndexError:
+        raise description
+
+
+def get_source_registry(tool_id):
+    """
+    Take a tool id and return the toolshed url if the id is a long id or a empty string if id is a short id
+    :param tool_id: Galaxy tool id
+    :return: toolshed url for a tool or empty string
+    :rtype: string
+    """
+    try:
+        return "/".join(tool_id.replace('repos', 'view', 1).split('/')[0:-2])
+    except ValueError:
+        log.warning("Not well formated tool id: {}".format(tool_id))
+        return ""
+
+
+def build_general_dict(tool_meta_data):
+    """
+    Extract informations from a galaxy json tool and return the general json in the biotools format
+    :param tool_meta_data: galaxy json tool
+    :return: biotools dictionary
+    :rtype: dictionary
+    """
+
+    gen_dict = {
+        'version': tool_meta_data[u'tool_version'],
+        'description': format_description(tool_meta_data[u'description']),
+        'uses': [{
+            "usesName": tool_meta_data[u'tool_id'],
+            "usesHomepage": "{0}?tool_id={1}".format(os.path.join('conf.galaxy_url', "root"),
+                                                     tool_meta_data[u'tool_id']),
+            "usesVersion": tool_meta_data[u'tool_version']
+        }],
+        'collection': '',
+        'sourceRegistry': get_source_registry(tool_meta_data[u'tool_id']),
+        'resourceType': ["Tool"],
+        'maturity': 'Stable',
+        'platform': [{u'term': 'Linux'}],
+        'interface': [{
+            'interfaceType': "Web UI",
+            'interfaceDocs': '',
+            'interfaceSpecURL': '',
+            'interfaceSpecFormat': ''
+        }],
+        'topic': [{
+            'uri': "http://edamontology.org/topic_0003",
+            'term': "EDAM label placeholder"
+        }],
+        'publications': {'publicationsPrimaryID': "None", 'publicationsOtherID': []},
+        'homepage': 'conf.galaxy_url',
+        'accessibility': '"Private" if conf.private else "Public"',
+        'mirror': [],
+        'canonicalID': '',
+        'tag': [],
+        'elixirInfo': {
+            'elixirStatus': '',
+            'elixirNode': ''
+        },
+        'language': [],
+        'license': '',
+        'cost': '',
+        'docs': {
+            'docsHome': '',
+            'docsTermsOfUse': '',
+            'docsDownload': '',
+            'docsCitationInstructions': ''
+        },
+        'credits': {
+            'creditsDeveloper': [],
+            'creditsContributor': [],
+            'creditsInstitution': [],
+            'creditsInfrastructure': [],
+            'creditsFunding': []
+        },
+        'contact': [{
+            'contactEmail': "conf.contactEmail",
```

Comment:
We could get this mandatory field from the galaxy config (either `error_email_to`, `admin_users` or `email_from` ), but maybe it would be better that consumers of this endpoint would fill in the contact details? This also depends on whether finally we also provide this endpoint on the toolshed?!